### DISCUSSION TITLE
support #tag in tag syntax

### DISCUSSION THREAD

**Discussion by delta4d** [zim-desktop-wiki/zim-desktop-wiki#1708] [Ideas]
can we support `#tag_name` also as tag?

In some other notebook/gtd system, it is usually using
- `@...` as context/person, such as `@home`, `@Alice`
- `#...` as some topic, such as `#clojure`, `#reading-list`

**Comment by sojusnik** [comment]
To have more options is not always better, especially when you can achieve the same results with existing things. Why not try to use a capital initial letter with "context/person" and a lower one with "topics", getting `@Home` `@Alice` and `@clojure` `@reading-list`.

**Comment by ecodiv** [comment]
Although I generally agree with 'more is not always better', the additional tag would bring it closer to e.g., todo.txt use of tags. See also #147  by Jaap.

**Comment by thecosmicyes** [comment]
I vote for this request.

**Comment by sojusnik** [comment]
Considering the integration of todo.txt, this indeed makes sense!

**Comment by jaap-karssenberg** [comment]
I think this needs to be rejected because it conflicts with linking to headings using "#heading" as is standard in HTML. Could an alternative be to allow tags to have a second non-alphanumeric tag. E.g. todo.txt projects could be reflected with @+project making the "+" part of the tag. Depending on preference you can use @_project or @#project 

(Using "+project" without leading "@" conflicts again with syntax for linking sub-pages - we are running out of free characters ...)

**Reply by casperl** [reply]
Agreed.  However there are a few long-term considerations. These are not really issues or feature requests but feedback on how I use Zim.

1) Markdown is the ultimate destination of articles I research, organize or write in Zim Notebooks and the less that conflicts with Markdown syntax, the better IMHO.   Out of habit I frequently use Markdown syntax while I am writing.

2) The @ symbol for tags is not ideal since I frequently paste text containing references to users on other social media platforms such as Twitter and all those superfluous @Rambo951 references are cluttering up my tags pane.  Perhaps using @@ for context style tags and @# for topic styled tags might work.  While the @ symbol for tags is not ideal, I can live with this issue.

3) As a wiki Zim is absolutely fantastic but over the long term I would like to see more inclusion of knowledge management features.  Since I already use a structured taxonomy on websites I maintain, I attempt to emulate that functionality in specific Notebooks through an appropriate document hierarchy.  There are cases when that is not functional (such as when more than one taxonomy is used) where I then create a wiki page where the taxonomical hierarchy is simulated by an indented bullet list and I then add links to appropriate pages in the Wiki if required — this then provides a special "index page" in a Notebook reflecting the taxonomy I use with links to the appropriate Notebook pages inserted into the taxonomy.  This is really a sophisticated type of site map reflecting either all or part of a Notebook that is [unfortunately] created manually for lack of a taxonomy functionality.

4) A problem I have is that most of the actual tags I use consist of two or more words and I commonly have to use the underscore character to combine multiple words into a tag.  AFAIK tags are limited to either single words, camel case or words separated with underscores and this type of tag definition is (unfortunately) not possible in Zim:

Tags: Tag One, Tag Two, Tag Three

**Comment by pfalcon** [comment]
> I think this needs to be rejected because it conflicts with linking to headings using "#heading" as is standard in HTML.

Hmm, yeah, but URL fragment markers don't appear on their own, they can be either part of link syntax or (perhaps) heading. So, they can be disambiguated by context. Of course, that's extra parsing complications, but ultimately, benefits depend on familiarity and interoperability with other tools. And Twitter popularized hashtags way too much to ignore them.

**Comment by pfalcon** [comment]
> Of course, that's extra parsing complications, but ultimately, benefits depend on familiarity and interoperability with other tools.

As a first step in surveying such other tools (beyond Twitter, Instagram, and the bunch), [Zettlr](https://www.zettlr.com/) uses hashtags (in the file/page body, yeah). Funnily, I can't immediately find online reference beyond https://docs.zettlr.com/en/reference/settings/#manage-tags , but a tutorial notebook included with it says:

> But creating links is not the only way to create relationships between notes. You can also use tags for this. Tags work exactly like hashtags on Twitter, so you can #create #hashtags #as #much #as #you #want! `Cmd/Ctrl`-clicking these will also start a search and will highlight all files that contain this tag. 

(Reminds me that I wanted to post https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1247).

**Comment by sojusnik** [comment]
> I think this needs to be rejected because it conflicts with linking to headings using "#heading" as is standard in HTML. Could an alternative be to allow tags to have a second non-alphanumeric tag. E.g. todo.txt projects could be reflected with @+project making the "+" part of the tag. Depending on preference you can use @_project or @#project
> 
> (Using "+project" without leading "@" conflicts again with syntax for linking sub-pages - we are running out of free characters ...)

I'm with @pfalcon on this. The familiarity of `#` and interoperability overweights the complexity of adding `#` as a second sign for tags. Can't imagine that people will use for tags a combination of `@` and other signs, as @jaap-karssenberg proposed with his examples like `@+project @_project @#project`. That's counterintuitive and I've never seen other apps doing that. Even [hyphens](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1250) are already too much. The only thing that IMHO could work is respecting upper- and lowercase letters in combination with `@`, as [mentioned](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/417#issuecomment-379882777) above. This worked well in previous versions, where the sorting in the tag cloud was case-sensitive, thus allowing to create a sort of two types of tags.

**Comment by hydrargyrum** [comment]
Conflicting with HTML URL fragments is a non-issue, if the hash is preceded by a URL-allowed character, then it's not a tag, if it's preceded by something else (e.g. a space or a newline), it is a tag: for example `foo.html#bar` is not a tag, `foo.html #bar` is a tag.

If referring to Markdown headings, Markdown requires whitespace after the hash, so it's obvious it's not a tag (which can't start with a space)

**Reply by introt** [reply]
In 0.74 `#foo` links to `id: foo` on the same page, so `#` won't doo
